tests/utilities/test_app.py

```diff
@@ -177,3 +221,107 @@ def disable_input(self) -> None:
 
     def stop_application_mode(self) -> None:
         pass
+
+
+# It seems that we have to give _way more_ time to `asyncio` on Windows in order to see our different awaiters
+# properly triggered when we pause our own "move clock forward" loop.
+# It could be caused by the fact that the time resolution for `asyncio` on this platform seems rather low:
+# > The resolution of the monotonic clock on Windows is usually around 15.6 msec.
+# > The best resolution is 0.5 msec.
+# @link https://docs.python.org/3/library/asyncio-platforms.html:
+ASYNCIO_EVENTS_PROCESSING_REQUIRED_PERIOD = 0.025 if WINDOWS else 0.002
+
+
+async def let_asyncio_process_some_events() -> None:
+    await asyncio.sleep(ASYNCIO_EVENTS_PROCESSING_REQUIRED_PERIOD)
+
+
+class ClockMock(_Clock):
+    def __init__(
+        self,
+        *,
+        ticks_granularity_fps: int = 60,
+    ):
+        self._ticks_granularity_fps = ticks_granularity_fps
+        self._single_tick_duration = 1.0 / ticks_granularity_fps
+        self._start_time = self._current_time = None
+        self._pending_sleep_events: list[tuple[float, asyncio.Event]] = []
+
+    def get_time(self) -> float:
+        if self._current_time is None:
+            self._start_clock()
+
+        # let's make the time advance _very_ slightly between 2 consecutive calls of this function,
+        # within the same order of magnitude than 2 consecutive calls to ` timer.monotonic()`:
+        self._current_time += 1.1e-06
+
+        return self._current_time
+
+    async def sleep(self, seconds: float) -> None:
+        event = asyncio.Event()
+        target_event_monotonic_time = self._current_time + seconds
+        self._pending_sleep_events.append((target_event_monotonic_time, event))
+        # Ok, let's wait for this Event
+        # (which can only be "unlocked" by calls to `move_clock_forward()`)
+        await event.wait()
+
+    async def move_clock_forward(self, *, seconds: float) -> tuple[float, int]:
```

Comment:
Suggest `"advance_clock"`

Does it need to be a keyword only argument? We generally exclusively use seconds for time.